src/PrestaShopBundle/Controller/Admin/Sell/CustomerService/CustomerThreadController.php

```
@@ -0,0 +1,225 @@
+            return $this->redirectToRoute('admin_customer_threads_view', [
+                'customerThreadId' => $customerThreadId,
+            ]);
+        }
+
+        if (!$replyToCustomerThreadForm->isValid()) {
+            foreach ($replyToCustomerThreadForm->getErrors(true) as $error) {
+                $this->addFlash('error', $error->getMessage());
+            }
+
+            return $this->redirectToRoute('admin_customer_threads_view', [
+                'customerThreadId' => $customerThreadId,
+            ]);
+        }
+
+        $data = $replyToCustomerThreadForm->getData();
+
+        try {
+            $this->getCommandBus()->handle(
+                new ReplyToCustomerThreadCommand((int) $customerThreadId, $data['reply_message'])
+            );
+
+            $this->addFlash(
+                'success',
+                $this->trans(
+                    'The message was successfully sent to the customer.',
+                    'Admin.Orderscustomers.Notification'
+                )
+            );
+        } catch (DomainException $e) {
+            $this->addFlash('error', $this->getErrorMessageForException($e, []));
+        }
+
+        return $this->redirectToRoute('admin_customer_threads_view', [
+            'customerThreadId' => $customerThreadId,
+        ]);
+    }
+
+    /**
+     * Update customer thread status
+     *
+     * @param int $customerThreadId
+     * @param string $newStatus
+     *
+     * @return RedirectResponse
+     */
+    public function updateStatusAction($customerThreadId, $newStatus)
+    {
+        try {
+            $this->getCommandBus()->handle(
+                new UpdateCustomerThreadStatusCommand((int) $customerThreadId, $newStatus)
+            );
+
+            $this->addFlash(
+                'success',
+                $this->trans('The status has been successfully updated.', 'Admin.Notifications.Success')
+            );
+        } catch (DomainException $e) {
```

We now need to catch all Exception because of https://github.com/PrestaShop/PrestaShop/pull/14239